icx/src/main.rs

```diff
@@ -387,28 +418,35 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
                 match result {
                     Ok(blob) => {
                         print_idl_blob(&blob, &t.output, &method_type)
-                            .map_err(|e| format!("Invalid IDL blob: {}", e))?;
+                            .context("Failed to print result blob")?;
                     }
                     Err(AgentError::TransportError(_)) => return Ok(()),
                     Err(AgentError::HttpError(HttpErrorPayload {
                         status,
                         content_type,
                         content,
                     })) => {
-                        eprintln!("Server returned an HTTP Error:\n  Code: {}", status);
+                        let mut error_message =
+                            format!("Server returned an HTTP Error:\n  Code: {}", status);
```

Comment:
Suggestion: add newlines at the end of these, like `eprintln` would have added, here and below:
```suggestion
                        let mut error_message =
                            format!("Server returned an HTTP Error:\n  Code: {}\n", status);
```